src/boot/SerializeEscaped.js

```diff
@@ -0,0 +1,155 @@
+   * the same object; otherwise, makes a new object and returns that.
+   * Original ordering of keys is not preserved in this new object.
+   */
+  (obj: InexactObject): InexactObject => {
+    const renamedEntries: $ElementType<$Call<typeof objectEntries, InexactObject>, number>[] = [];
+    const untouchedEntries: $ElementType<$Call<typeof objectEntries, InexactObject>, number>[] = [];
+    for (const entry of objectEntries(obj)) {
+      const [key, _] = entry; /* eslint-disable-line no-unused-vars */
+      const newName = renameFn(key);
+      if (key !== newName) {
+        entry[0] = newName; // *elements* of a $ReadOnlyArray may be mutable
+        renamedEntries.push(entry);
+      } else {
+        untouchedEntries.push(entry);
+      }
+    }
+    if (renamedEntries.length > 0) {
+      return {
+        // Object.fromEntries is added in facebook/flow@f92231cf1,
+        // released in v0.96.0. Until we use that, $FlowFixMe.
+        ...Object.fromEntries(untouchedEntries),
+        // $FlowFixMe
+        ...Object.fromEntries(renamedEntries),
+      };
+    } else {
+      return obj;
+    }
+  };
+
+const escapeKeys = makeObjectKeyRenamer(escape);
+const unescapeKeys = makeObjectKeyRenamer(unescape);
+
+/**
+ * `Serialize.immutable` from `remotedev-serialize`, but escaped.
+ *
+ * Always use this wrapper, to protect against a particular DoS.
+ *
+ * It has exactly the same API, but this implementation ensures that
+ * custom and default replacers/revivers also escape the
+ * `__serializedType__` key, which is used as an identifier, not as
+ * data.
+ *
+ * If we don't escape inputs, a user could maliciously compose a piece
+ * of data with this special key and trigger hidden functionality.
+ */
+/* eslint-disable-next-line no-underscore-dangle */
+const _immutable: typeof Serialize.immutableDangerouslyMissingWarning = (
+  // Our `remotedev-serialize` libdef has `immutable: any`, while the
+  // DefinitelyTyped libdef has `immutable: typeof Immutable`; see
+  // https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md#dont-import-types-from-other-libdefs.
+  // We can do better here, outside the libdef, so, do.
+  immutable: typeof Immutable,
+  refs?: Refs | null,
+  customReplacer?: Replacer,
+  customReviver?: Reviver,
+) => {
+  const escapingReplacer: Replacer = (key, value, defaultReplacer) => {
+    const safeValue = value !== null && typeof value === 'object' ? escapeKeys(value) : value;
+
+    return customReplacer !== undefined
+      ? customReplacer(key, safeValue, defaultReplacer)
+      : defaultReplacer(key, safeValue);
+  };
+  const unescapingReviver: Reviver = (key, value, defaultReviver) => {
+    const result = customReviver
+      ? customReviver(key, value, defaultReviver)
+      : defaultReviver(key, value);
+
+    const safeResult =
+      result !== null && typeof result === 'object' ? unescapeKeys(result) : result;
+    return safeResult;
+  };
```

Comment:
Do you think `escapingReplacer` and `unescapingReviver` need some more comments? The `key` param is not transformed on its way into `custom`- and `default`- `Replacer` and `Reviver`. Rather, keys of an object at the `value` param are transformed. The functions ask you to return some transformation of the `value` param, and you can optionally involve the `key` param in that transformation.

These replacers and revivers are passed to JSON.stringify and JSON.parse as their second parameter, so it might be good to link to MDN ([stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#The_replacer_parameter), [parse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#Using_the_reviver_parameter)) from a comment in the code?

I followed it all the way down to JSON.stringify and JSON.parse via:

1. node_modules/remotedev-serialize/immutable/serialize.js, where there's some logic to default to replacers/revivers defined by `remotedev-serialize`
2. node_modules/remotedev-serialize/immutable/index.js, where the custom-or-default replacers/revivers are passed to `jsan.stringify` [sic] and `jsan.parse` [sic].
3. node_modules/jsan/lib/index.js, where the replacer/reviver is passed to `JSON.stringify` and `JSON.parse`.